app/vue/src/client/preview/index.ts

```diff
@@ -0,0 +1,127 @@
+/* eslint-disable prefer-destructuring */
+import Vue from 'vue';
+import { start } from '@storybook/core/client';
+import {
+  ClientStoryApi,
+  StoryFn,
+  DecoratorFunction,
+  StoryContext,
+  Loadable,
+} from '@storybook/addons';
+
+import './globals';
+import { IStorybookSection, StoryFnVueReturnType } from './types';
+
+import render, { VALUES } from './render';
+import { extractProps } from './util';
+
+export const WRAPS = 'STORYBOOK_WRAPS';
+
+function prepare(rawStory: any, innerStory?: any): StoryFnVueReturnType {
+  let story = rawStory;
+  // eslint-disable-next-line no-underscore-dangle
+  if (!story._isVue) {
+    if (typeof story === 'string') {
+      story = { template: story };
+    }
+    if (innerStory) {
+      story.components = { ...(story.components || {}), story: innerStory };
+    }
+    story = Vue.extend(story);
+  } else if (story.options[WRAPS]) {
+    return story;
+  }
+
+  return Vue.extend({
+    // @ts-ignore // some Vue expert needs to look at this
+    [WRAPS]: story,
+    [VALUES]: { ...(innerStory ? innerStory.options[VALUES] : {}), ...extractProps(story) },
+    functional: true,
+    render(h, { data, parent, children }) {
+      return h(
+        story,
+        {
+          ...data,
+          // @ts-ignore // some Vue expert needs to look at this
+          props: { ...(data.props || {}), ...parent.$root[VALUES] },
```

Comment:
`VALUES` is a reactive data property on the $root vue instance. see the comment in `render.ts` for an explanation. (since this is a functional component, we cant do `this.$root`, but have to use the parents `$root`)